nim: 1.6.8 -> 1.6.10

shassard opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

shassard committed:
nim: 1.6.8 -> 1.6.10

trofi commented:
Result of `nixpkgs-review pr 202983` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>hottext</li>
  </ul>
</details>
<details>
  <summary>31 packages built:</summary>
  <ul>
    <li>mosdepth</li>
    <li>nim (nimPackages.nim)</li>
    <li>nim-unwrapped</li>
    <li>nimPackages.bumpy</li>
    <li>nimPackages.c2nim</li>
    <li>nimPackages.flatty</li>
    <li>nimPackages.hts-nim</li>
    <li>nimPackages.jsonschema</li>
    <li>nimPackages.jsony</li>
    <li>nimPackages.nim_builder</li>
    <li>nimPackages.pixie</li>
    <li>nimPackages.rocksdb</li>
    <li>nimPackages.sass</li>
    <li>nimPackages.sdl2</li>
    <li>nimPackages.snappy</li>
    <li>nimPackages.spry</li>
    <li>nimPackages.spryvm</li>
    <li>nimPackages.stew</li>
    <li>nimPackages.tempfile</li>
    <li>nimPackages.ui</li>
    <li>nimPackages.vmath</li>
    <li>nimPackages.zippy</li>
    <li>nimble-unwrapped</li>
    <li>nimlsp</li>
    <li>nimmm</li>
    <li>nitch</li>
    <li>nitter</li>
    <li>nrpl</li>
    <li>swaycwd</li>
    <li>tridactyl-native</li>
    <li>vimPlugins.fruzzy</li>
  </ul>
</details>

Commit:
Nim: resurrect bootstrap compiler

Building the bootstrap and the unwrapped compiler in seperate
derivations is still worthwhile because if we patch the compiler
only need to rebuild the second of the two.
